commands/Tools/Step 3 - Modify schematisation/control_structures.py

```diff
@@ -0,0 +1,210 @@
+            id_nr = weir_id[0]
+            self.dockwidget_controlled_structures.\
+                combobox_input_structure_id.addItem(str(id_nr))
+
+    def save_control_structure(self):
+        """
+        Function to save the created control in the designated table.
+        Currently only works for table control.
+        """
+        control_type = TABLE_CONTROL
+        # Get the model
+        db_key = self.dockwidget_controlled_structures\
+            .combobox_input_model.currentText()
+        db_entry = self.databases[db_key]
+
+        _db_settings = db_entry['db_settings']
+
+        if db_entry['db_type'] == 'spatialite':
+            host = _db_settings['db_path']
+            db_settings = {
+                'host': host,
+                'port': '',
+                'name': '',
+                'username': '',
+                'password': '',
+                'schema': '',
+                'database': '',
+                'db_path': host,
+            }
+        else:
+            db_settings = _db_settings
+            db_settings['schema'] = 'public'
+        control_structure = ControlledStructures(
+            flavor=db_entry['db_type'])
+        control_structure.start_sqalchemy_engine(db_settings)
+        # Get the variables as input for the v2_control_table
+        if control_type == TABLE_CONTROL:
+            measure_value = self.dockwidget_controlled_structures\
+                .tablewidget_input_rule_table_control.item(0, 0).text()
+            action_value = self.dockwidget_controlled_structures\
+                .tablewidget_input_rule_table_control.item(0, 1).text()
+            action_table = '{0};{1}'.format(measure_value, action_value)
+            if self.dockwidget_controlled_structures\
+                    .combobox_input_rule_operator.currentText()\
+                    == BOTTOM_UP:
+                measure_operator = RULE_OPERATOR_BOTTOM_UP
+            else:
+                measure_operator = RULE_OPERATOR_TOP_DOWN
+            target_id = self.dockwidget_controlled_structures\
+                .combobox_input_structure_id.currentText()
+            if self.dockwidget_controlled_structures\
+                    .combobox_input_structure_table.currentText()\
+                    == V2_WEIR_VIEW_TABLE:
+                target_type = V2_WEIR_TABLE
+                action_type = ACTION_TYPE_SET_CREST_LEVEL
+            else:
+                target_type = ""
+                action_type = ''
+            if self.dockwidget_controlled_structures\
+                    .combobox_input_measuring_point_field.currentText()\
+                    == INITIAL_WATERLEVEL:
+                measure_variable = WATERLEVEL
+            else:
+                measure_variable = ""
+            with control_structure.engine.connect() as con:
+                rs = con.execute(
+                    '''SELECT MAX(id) FROM v2_control_table;'''
+                )
+                max_id_control_table = rs.fetchone()[0]
```

Comment:
what happens if the sql statement yields nothing? Or does ``MAX(id)`` always returns an integer? Is it possible that the index lookup throws an index error? Maybe good to add some comments

Reply:
Good idea.
If the the sql statement yields nothing, it returns 'None' and raises an Type Error when adding 1. 
(TypeError: unsupported operand type(s) for +: 'NoneType' and 'int').

I'll describe it in comments.